### PR TITLE
Fix: Fix resize logic during preload and opening of thumbnails

### PR DESCRIPTION
**Pull request by jeremypress**
- [x] unit tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 1]
```diff
@@ -667,6 +667,7 @@ class DocBaseViewer extends BaseViewer {
                 if (this.shouldThumbnailsBeToggled()) {
                     this.rootEl.classList.add(CLASS_BOX_PREVIEW_THUMBNAILS_OPEN);
                     this.emit(VIEWER_EVENT.thumbnailsOpen);
+                    this.resize();
```
Comment (jeremypress) [inline src/lib/viewers/doc/DocBaseViewer.js]: Now that we aren't transitioning on initial load, a resize without a timeout will sufficiently resize the preload or the doc depending on what's ready.

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 2]
```diff
@@ -720,7 +721,7 @@ class DocBaseViewer extends BaseViewer {
      * @return {void}
      */
     resize() {
-        if (!this.pdfViewer || !this.pdfViewer.pageViewsReady) {
+        if (!this.pdfViewer || !this.somePageRendered) {
```
Comment (jeremypress) [inline src/lib/viewers/doc/DocBaseViewer.js]: `this.pdfViewer.pageViewsReady` was updating way after a page was rendered. This new check waits for a rendered event from PDF.js so we know the pdfViewer is inter-actable.

File: src/lib/viewers/doc/DocPreloader.js [hunk 3]
```diff
@@ -268,11 +268,19 @@ class DocPreloader extends EventEmitter {
      * @return {void}
      */
     resize() {
-        if (!this.pdfData || !this.preloadEl) {
+        if (!this.preloadEl || (!this.pdfData && !this.imageEl)) {
```
Comment (jeremypress) [inline src/lib/viewers/doc/DocPreloader.js]: This was causing the resize issue with the preview sidebar. If an image didn't have exif data burnt into it, we wouldn't be able to resize and would see that overlap.

File: src/lib/viewers/doc/_docBase.scss [hunk 4]
```diff
@@ -113,7 +113,8 @@ $thumbnail-sidebar-width: 226px;
     @include bp-breakpoint-medium {
         $bp-thumbnails-animation: 300ms cubic-bezier(.4, 0, .2, 1);
 
-        &.bp-loaded {
+        &.bp-loaded.bp-thumbnails-open-active,
```
Comment (jeremypress) [inline src/lib/viewers/doc/_docBase.scss]: Now we'll only apply the transition to bp content if the active classes are present. Those classes only get added in the user generated toggle case.

**Pull request merged**